velox/type/DecimalUtil.h

```diff
@@ -203,32 +204,57 @@ class DecimalUtil {
     return static_cast<TOutput>(rescaledValue);
   }
 
-  /// Rescales a double value to decimal value of given precision and scale. The
-  /// output is rescaled value of int128_t or int64_t type. Returns error status
-  /// if fails.
-  template <typename TOutput>
-  inline static Status
-  rescaleDouble(double value, int precision, int scale, TOutput& output) {
+  /// Rescales a floating point value to decimal value of given precision and
+  /// scale. The output is rescaled value of int128_t or int64_t type. Returns
+  /// error status if fails.
+  template <typename TIntput, typename TOutput>
+  inline static Status rescaleFloatingPoint(
+      TIntput value,
+      int precision,
+      int scale,
+      TOutput& output) {
     if (!std::isfinite(value)) {
```

Comment:
Do we need to check for NaN here as well?

Reply:
NaN is covered here as std::isfinite returns false for NaN.